src/oops/SOLID/DependencyInversionPrinciple/before/notification/EmployeeEmailNotifier.java

```diff
@@ -9,8 +9,10 @@
 server data; The current configuration properties will not be
 valid.
  */
-public class EmailSender {
-    public static void notify(Employee employee){
+public class EmployeeEmailNotifier implements Notifier<Employee>{
+    
+	@Override
+	public void notify(Employee employee){
```

Comment:
This could be nit-picky, but I was wondering if this method should care whether we are sending mail to employee or a contractor. What I mean is, method signature can be,

`void notify(final String email, final String subject, final String body);`

Not necessarily a good design, but just a thought.

Reply:
As a matter of fact, you could also create a type `Email` to have members as `Subject`, `To` and `Body` to simplify the method signature and data handling. I can do that change if you think its better. What do you say?

Reply:
What you are suggesting is `Microtype` design pattern. Give it a read, it is very fascinating concept.

I am not sure what would be a good design here. It was just a thought. May be @kshitijmishra23 can helps us here?

Reply:
It appears to be a good suggestion, what @Kalyan7767 has mentioned above. 
@AmeyKamat Do you have any reference for the Microtype design pattern. I couldn't find one.

Reply:
`Microtype` is basically a classwrapper over primitive data types. An examples for microtype is autoboxable classes in java such as `Integer`, `Double` and for that matter even `String`.

Check out this link:

https://codebox.net/pages/microtypes-in-java

Reply:
Thanks for the reference :)